Cargo.toml

```diff
@@ -15,6 +15,7 @@ embedded-hal = "0.2"
 stm32f4xx-hal = { version = "0.8" , optional = true }
 smoltcp = { version = "0.6.0", default-features = false, features = ["proto-ipv4", "proto-ipv6", "socket-icmp", "socket-udp", "socket-tcp", "log", "verbose", "ethernet"], optional = true }
 log = "0.4"
+cortex-m-rtic = "0.5.3"
```

Comment:
Why is this a hard general dependency?

Reply:
Thanks! It shouldn't be a required dependency for just the driver alone. I fixed this with 14a75c904631219f1143bbc4c5cc9f01ed3ccebd.

Reply:
Forced-pushed branch to fix some messy code and add delays (see #3). The corresponding commit now is f6112bf3fc3324264dfc2649aed9a332a43a7f6b.